include/engine/plugins/plugin_base.hpp

```diff
@@ -242,17 +279,31 @@ class BasePlugin
                     phantom_node_pairs[i] =
                         facade.NearestPhantomNodeWithAlternativeFromBigComponent(
                             parameters.coordinates[i],
-                            *parameters.radiuses[i],
+                            max_radius_when_bearings != EngineConfig::UNLIMITED
+                                ? std::min(*parameters.radiuses[i], max_radius_when_bearings)
+                                : *parameters.radiuses[i],
                             parameters.bearings[i]->bearing,
                             parameters.bearings[i]->range);
                 }
                 else
                 {
-                    phantom_node_pairs[i] =
-                        facade.NearestPhantomNodeWithAlternativeFromBigComponent(
-                            parameters.coordinates[i],
-                            parameters.bearings[i]->bearing,
-                            parameters.bearings[i]->range);
+                    if (max_radius_when_bearings == EngineConfig::UNLIMITED)
+                    {
```

Comment:
same question as above, only `==` is correct here. Still, why the split into two functions? Why not call it with UNLIMITED?